lib/rules/no-implicit-injections.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const kebabCase = require('lodash.kebabcase');
+const camelCase = require('lodash.camelcase');
+
+const defaultServiceConfig = { serviceName: 'store', moduleNames: ['Route', 'Controller'] };
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function fixService(fixer, currentClass, serviceInjectImportName, servicePropertyName) {
+  const serviceInjectPath = kebabCase(servicePropertyName);
+
+  return fixer.insertTextBefore(
+    currentClass.node.body.body[0],
+    `@${serviceInjectImportName}('${serviceInjectPath}') ${servicePropertyName};\n`
+  );
+}
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce usage of implicit service injections',
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-implicit-injections.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          services: {
+            minItems: 1,
+            type: 'array',
+            items: {
+              type: 'object',
+              default: [defaultServiceConfig],
+              properties: {
+                serviceName: {
+                  type: 'string',
```

Comment:
Let's also use `minLength: 1` for strings.

Reply:
Updated and marked `required` for the values in these objects too.